eras/alonzo/test-suite/src/Test/Cardano/Ledger/Alonzo/AlonzoEraGen.hs

```diff
@@ -433,12 +426,26 @@ instance Mock c => EraGen (AlonzoEra c) where
         Nothing -> storageCost 0 pp script
       else storageCost 0 pp script
 
-  genEraDone pp tx =
+  -- For some reason, the EraGen generators occasionally generate an extra script witness.
+  --    There is some evidence that this arises because the script hash appears as the PolicyId
+  --    in a Value. But that is not been verified. Regardless of the cause, we can fix this by
+  --    discarding the trace. Note that this is failure to generate a "random" but valid
+  --    transaction. Discarding the trace adjust for this inadequacy in the generation process.
+  --    This only appears in the Alonzo era, so this "fix" is applied here, in the genEraDone
+  --    method of the EraGen class in the (AlonzoEra c) instance.
+  genEraDone utxo pp tx =
     let theFee = tx ^. bodyTxL . feeTxBodyL -- Coin supplied to pay fees
         minimumFee = getMinFeeTx @(AlonzoEra c) pp tx
+        neededHashes = snd <$> (scriptsNeeded utxo tx)
+        oldScriptWits = (tx ^. witsAlonzoTxL . scriptAlonzoTxWitsL)
+        newWits = Map.filterWithKey (\k _ -> List.elem k neededHashes) oldScriptWits
      in if minimumFee <= theFee
-          then pure tx
-          else myDiscard "MinFeee violation: genEraDne: AlonzoEraGen.hs"
+          then
+            ( if not (oldScriptWits == newWits)
+                then myDiscard "Random extra scriptwitness: genEraDone: AlonzoEraGen.hs"
+                else pure tx
+            )
```

Comment:
Redundant parenthesis and unnecessary negation
```suggestion
          then
              if oldScriptWits /= newWits
                then myDiscard "Random extra scriptwitness: genEraDone: AlonzoEraGen.hs"
                else pure tx
```

Reply:
done